### PR TITLE
[PROPOSAL] - feature/registrator-module

### PR DESCRIPTION
**Pull request by JozefFlakus**
**1 AIM:**

Provide expandable interface for registering model schemas. By analyzing schema structure we will be able to extract and compose schema metadata which we can use as an interface for model administrator panels in `@barista/admin`.

**2 AIM:**

The second aim is to create an interface which is _expandable_. It means that it should be able to deal with a variety of other database engines (_future plans)_. For MVP purpose we need an **MongoDB** adapter.

### CODE REVIEW

File: src/registrator/adapters/mongoose.adapter.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import { connection, connect, model, Schema } from 'mongoose';
+import { Metadata, IMetadata, IMetadataField } from './mongoose.model';
+import { Adapter, AdapterConfig, MongooseAdapterConfig  } from '../registrator.interface';
+
+const extractFieldMetadata = (field: any): IMetadataField => ({
```
Comment (pdomaleczny) [inline src/registrator/adapters/mongoose.adapter.ts]: Can we destruct these fields in parameters section `{ path, isRequired, instance ...} : any` ?

**Comment by JozefFlakus** [comment]
UPDATE:

REGISTRATOR:
- added dedicated `env.ts` file for setting up package environment variables (only needed for contributors testing purposes)
- `extractMetadata()` - filtered fields: `_id`, `__v`, `updatedAt`, `createdAt` which are added via mongoose
-  `metadata.controller.ts` - removed from returned JSON fields: `_id`, `__v` which were not needed for end users
- `metadata.controller.ts` - `'api/metadata/:name'` now works with lowercased model name

GENERATOR:
- prepared generator module for working registrator module feature (still not added `@barista/core` to the `package.json` file)
- connected `bluebird` promise library with mongoose (it shouldn't throw mongoose warnings anymore)
- tested `User` model registration - it works very well but still need some fixes and improvements :)

**Comment by JozefFlakus** [comment]
*UPDATE:*

New attribute for mongodb Schema definitions: `inputType` - handy for automatic building model forms. Available values:
- `text`,
- `password`,
- `number`,
- `textarea`,

Example use case:
```
...
password: { type: String, select: false, inputType: 'password' },
...
```

**Comment by JozefFlakus** [comment]
**UPDATE:**

Auto register model endpoints for admin panel purposes:
```
GET /api/admin/{model}/             - get all records
GET /api/admin/{model}/:id         - get specific record
PUT /api/admin/{model}/:id         - update specific record
POST /api/admin/{model}/           - create record
DELETE /api/admin/{model}/:id   - remove specific record 
```

**Comment by JozefFlakus** [comment]
@pdomaleczny please PTAL, right now you should see the full overview of this feature module

**Pull request merged**